services/py-api/src/server/config/server_config.py

```diff
@@ -28,19 +35,18 @@ def start() -> None:
     server_config = load_server_config()
 
     if server_config.ENV in ("PROD", "DEV", "LOCAL", "TEST"):
-        configure_app_logger(server_config.ENV)
-
         err = ping_db()
         if err:
             raise RuntimeError(err.err_value)
 
-        # If the ENV is PROD or TEST we don't want to have hot reloading
         run(
             app="src.server.app_entrypoint:app",
             host=server_config.ADDRESS,
             port=server_config.PORT,
-            reload=server_config.ENV == "LOCAL",
-            log_config=get_uvicorn_logger(server_config.ENV),
+            # Reloading is intended only for local development. Simulating a PROD/DEV env will result in multiple
+            # workers spawned
+            reload=server_config.ENV == "LOCAL" and server_config.SIMULATING_PROD_ENV is False,
```

Comment:
What's the difference between pointing `ENV` to `"PROD"` and setting `SIMULATING_PROD_ENV` to True?

I don’t believe an additional variable is necessary to simulate the environment. For accurate simulation, it’s best to closely resemble the original environment. Adding an extra variable here may simply introduce more factors that could cause divergence from the intended conditions.

If you just want to be able to control the workers, you can extract the count as an environment variable.

`WORKERS_COUNT=0` for local environments

Reply:
I thought about your proposal. But it doesn't take into consideration our current logging situation. Before the introduction of the `SIMULATING_PROD_ENV`, we sent the logs to a file only if `ENV` is "DEV" or "PROD". But In the `Dokcerfile` we have a check which says if `ENV=LOCAL` don't check the validity of the self-signed certs. In the previous setup (before introducing `SIMULATING_PROD_ENV`), If you change `ENV=DEV`  and run the container locally, the logs will be written to a file, but the certs validation will fail and the container will be marked unhealthy :D. Also, the workers' count is calculated on demand via `(2 x $num_cores) + 1`, and I am not sure if we want to change this. And if reloading is enabled the `workers` flag is ignored, making it a bit of a mess. That's why I introduced the variable.